pkg/kubelet/secret/secret_manager.go

```diff
@@ -201,8 +201,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 5 minutes (which is a lot, but...)
+	ttl := 5 * time.Minute
```

Comment:
@wojtek-t your suggestion sounds reasonable (although adding yet another flag in kubelet doesn't excite me)

> The problem with that is that it will be configured on cluster startup (I guess), thus it will not work if someone will scale the cluster from 1 to 5000. But maybe that's not that big deal.

That could potentially be resolved by dynamic kubelet configs (#27980).

> many of the things involved in volume sync don't scale to 5000 nodes (like cloudprovider storage API calls). Would you envision this flag controlling the entire volume sync loop?

@liggitt I don't think this flag could/should capture the behavior of the volume sync. Maybe that should be addressed separately.